ls1intum/Artemis
# [Bugfix] - Added custom validator for Email Pattern

### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, editor, instructor, admin) on the test server https://artemistest2.ase.in.tum.de.

### Motivation and Context


This is the follow-up PR of https://github.com/ls1intum/Artemis/pull/3553 to provide a better solution to the invalid email patterns problem.

### Description

This PR adds an additional form validator to ensure that the inserted patterns are valid. 

### Steps for Testing


1. Log in to Artemis
2. Navigate to Organization Management
3. Add or edit an organization
4. Try to insert an invalid email pattern in the form (save button should be disabled and a message should appear)

### Test Coverage




### Screenshots


![Screenshot from 2021-06-17 15-14-35](https://user-images.githubusercontent.com/45761921/122403556-c140a400-cf7e-11eb-981b-741b578924a1.png)
